src/gsumm.c

```diff
@@ -86,8 +86,7 @@ SEXP gforce(SEXP env, SEXP jsub, SEXP o, SEXP f, SEXP l, SEXP irowsArg) {
   // TODO: enable stress-test mode in tests only (#3205) which can be turned off by default in release to decrease overhead on small data
   //       if that is established to be biting (it may be fine).
   if (nBatch<1 || batchSize<1 || lastBatchSize<1) {
-    error(_("Internal error: nrow=%d  ngrp=%d  nbit=%d  shift=%d  highSize=%d  nBatch=%d  batchSize=%d  lastBatchSize=%d\n"),  // # nocov
-           nrow, ngrp, nb, shift, highSize, nBatch, batchSize, lastBatchSize);                                              // # nocov
+    INTERNAL_ERRORF("nrow=%d  ngrp=%d  nbit=%d  shift=%d  highSize=%"PRIu64"  nBatch=%"PRIu64"  batchSize=%"PRIu64"  lastBatchSize=%"PRIu64, nrow, ngrp, nb, shift, (uint64_t)highSize, (uint64_t)nBatch, (uint64_t)batchSize, (uint64_t)lastBatchSize); // # nocov                                            // # nocov
```

Comment:
Not sure why this has been passing through now. `highSize`, `nBatch`, `batchSize`, `lastBatchSize` are all `size_t`, so `%d` shouldn't work.

I couldn't find what format should work for `size_t`, I cribbed this cast to `unit64_t` from `fread.c`, happy to change